### PR TITLE
Use enum for approximate suggestions

### PR DESCRIPTION
**Pull request by Manishearth**
r? @nrc @killercup

### CODE REVIEW

File: src/librustc_errors/lib.rs [hunk 1]
```diff
@@ -56,6 +56,14 @@ mod lock;
 
 use syntax_pos::{BytePos, Loc, FileLinesResult, FileMap, FileName, MultiSpan, Span, NO_EXPANSION};
 
+#[derive(Copy, Clone, Debug, PartialEq, Hash, RustcEncodable, RustcDecodable)]
+pub enum SuggestionApproximate {
```
Comment (zackmdavis) [inline src/librustc_errors/lib.rs]: I feel like this should have a different name given that it seems to include non-approximate (`MachineApplicable`) suggestions. `SuggestionDefiniteness`?

Comment (killercup) [reply src/librustc_errors/lib.rs]: I see what you mean. My first suggestion would've been SuggestionApplicability, though. (Let the bikeshedding begin)

Comment (zackmdavis) [reply src/librustc_errors/lib.rs]: I like `Applicability`

Comment (Manishearth) [reply src/librustc_errors/lib.rs]: Updated

**Pull request merged**